### PR TITLE
Image builder using local RPM packages

### PR DESCRIPTION
**Pull request by ggiguash**
Implement a procedure for building microshift installer ISO using the locally built (for microshift) & downloaded (for OCP) RPM packages.

The build procedure amounts to running the following scripts from scripts/image-builder directory:
- configure.sh - One time build tool installation and system configuration
- cleanup.sh - Delete all the previous artifacts, podman & osbuilder images to conserve disk space
- make rpm - Build MicroShift locally and prepare its RPM packages
- build.sh - Build the _builds/microshift-installer.x86_64.iso

[USHIFT-54](https://issues.redhat.com//browse/USHIFT-54)

### CODE REVIEW

File: scripts/image-builder/blueprint_v0.0.1.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+groups = []
+
+# MicroShift and dependencies
+
+[[packages]]
+name = "microshift"
+version = "*"
+
+[[packages]]
+name = "conntrack-tools"
```
Comment (fzdarsky) [inline scripts/image-builder/blueprint_v0.0.1.toml:15]: `conntrack-tools` are a dependency of microshift.rpm now, so this entry should no longer be needed.

File: scripts/image-builder/build.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    done
+
+    local tend=$(date +%s)
+    echo "$(date +'%Y-%m-%d %H:%M:%S') ${status} - elapsed $(( (tend - tstart) / 60 )) minutes"
+    
+    if [ "${status}" = "FAILED" ]; then
+        download_image ${uuid} 1
+        echo "Blueprint build has failed. For more information, review the downloaded logs"
+        exit 1
+    fi
```
Comment (fzdarsky) [inline scripts/image-builder/build.sh:33]: This contains good improvements over `https://github.com/redhat-et/microshift-demos/blob/main/_lib/build.sh`. Would it make sense to contribute them back there, too?

File: scripts/image-builder/build.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+title "Downloading local OpenShift and MicroShift repositories"
+# Copy microshift local RPM packages
+rm -rf microshift-local 2>/dev/null || true
+cp -TR "${ROOTDIR}/../../packaging/rpm/_rpmbuild/RPMS" microshift-local
+createrepo microshift-local >/dev/null
+
+# Download openshift local RPM packages
+rm -rf openshift-local 2>/dev/null || true
+reposync -n -a x86_64 --download-path openshift-local --repo=rhocp-4.10-for-rhel-8-x86_64-rpms >/dev/null
```
Comment (mangelajo) [inline scripts/image-builder/build.sh:100]: what's this for?, I guess it's may be part of frank/ricky's isntructions?

File: scripts/image-builder/installer.toml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+name = "microshift-installer"
```
Comment (mangelajo) [inline scripts/image-builder/installer.toml:1]: What's microshift-installer?

**Review by mangelajo — COMMENTED** [review]
some silly questions, amazing

**Comment by fzdarsky** [comment]
Running the `build.sh`, it builds the microshift-container blueprint correctly, but then fails when dep'solving the microshift-installer blueprint with:
```
ERROR: BlueprintsError: microshift-installer: DNF error occured: RepoError: Error occurred when setting up repo: Failed to download metadata for repo '0': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

Journal shows:

```
May 30 17:14:16 rhel8.localdomain osbuild-composer[28192]: 2022/05/30 17:14:16 GET /api/v1/blueprints/depsolve/microshift-installer
May 30 17:14:16 rhel8.localdomain dnf-json[53115]: Errors during downloading metadata for repository '0':
May 30 17:14:16 rhel8.localdomain dnf-json[53115]:   - Curl error (58): Problem with the local SSL certificate for https://cdn.redhat.com/content/dist/rhel8/8.6/x86_64/baseos/os/repodata/repomd.xml [could not load PEM client certificate, OpenSSL error error:02001002:system library:fopen:No such file or directory, (no key found, wrong pass phrase, or wrong file>
May 30 17:14:16 rhel8.localdomain dnf-json[53115]: 2022.05.30 17:14:16 dnf-json.INFO: clean unused folders
```

It's likely because I'm building from a non-clean system. Will retry this tomorrow.

**Comment by ggiguash** [comment]
@fzdarsky, the error about the certificates is due to older OS you're probably using for running osbuild-composer. I'm not longer copying rhel8.json files with the right certificates because they got fixed in RHEL 8.5/8.6. 
How should we proceed? Can you update your system?

**Comment by fzdarsky** [comment]
> @fzdarsky, the error about the certificates is due to older OS you're probably using for running osbuild-composer. I'm not longer copying rhel8.json files with the right certificates because they got fixed in RHEL 8.5/8.6. How should we proceed? Can you update your system?

It is already a RHEL8.6 system with the rhel8*.json files removed. Let me reprovision and try again.

**Pull request merged**